docs/examples/notebooks/spectrogram.ipynb

```diff
@@ -0,0 +1,273 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "c28810ab",
+   "metadata": {},
+   "source": [
+    "# Audio spectrogram in rocAL \n",
+    "\n",
+    "This example presents a simple rocAL pipeline that loads and decodes audio data along with the calculation of a spectrogram. We use MIVISIONX-data which contains audio data samples in wav format. Illustrated below how to create a pipeline, set_outputs, build, run the pipeline and enumerate over the results.\n",
+    "\n",
+    "## Reference implementation\n",
+    "\n",
+    "To verify the correctness of rocAL's implementation, we will compare it against librosa.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "862b24ff",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from __future__ import absolute_import\n",
+    "from __future__ import division\n",
+    "from __future__ import print_function\n",
+    "\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "import librosa.display\n",
+    "import librosa as librosa\n",
+    "\n",
+    "import numpy as np\n",
+    "import torch\n",
+    "torch.set_printoptions(threshold=10_000)\n",
+    "\n",
+    "import amd.rocal.types as types\n",
+    "import amd.rocal.fn as fn\n",
+    "from amd.rocal.pipeline import Pipeline, pipeline_def\n",
+    "from amd.rocal.plugin.pytorch import ROCALAudioIterator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "aef16554",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def show_spectrogram(spec, title, sr, hop_length, y_axis='log', x_axis='time'):\n",
+    "    librosa.display.specshow(\n",
+    "        spec, sr=16000, y_axis=y_axis, x_axis=x_axis, hop_length=hop_length)\n",
+    "    plt.title(title)\n",
+    "    plt.colorbar(format='%+2.0f dB')\n",
+    "    plt.tight_layout()\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0a7ab152",
+   "metadata": {},
+   "source": [
+    "## Librosa implementation\n",
+    "Librosa provides an API to calculate the STFT, producing a complex output (i.e. complex numbers). It is then trivial to calculate the power spectrum from the complex STFT by the following.\n",
+    "Here we load and decoder the audio file and applied spectrogram to it using librosa.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "84f22a01",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# Modify the path of MIVISIONX-data as needed\n",
+    "data_path = \"/data/MIVisionX-data/rocal_data/audio//wav/19-198-0000.wav\"\n",
```

Comment:
I suggest getting the path from an environment variable instead of hardcoded path.
For e.g ROCAL_DATA_PATH = "path for rocal_data"